src/main/java/mSearch/filmlisten/FilmlisteLesen.java

```diff
@@ -73,39 +67,8 @@ public static void setWorkMode(WorkMode mode) {
         workMode = mode;
     }
 
-    private InputStream getInputStreamForLocation(String source) throws IOException, URISyntaxException {
-        InputStream in;
-        long size = 0;
-        final URI uri;
-        if (source.startsWith("http")) {
-            uri = new URI(source);
-            //remote address for internet download
-            HttpURLConnection conn = (HttpURLConnection) uri.toURL().openConnection();
-            conn.setConnectTimeout(TIMEOUT);
-            conn.setReadTimeout(TIMEOUT);
-            conn.setRequestProperty("User-Agent", Config.getUserAgent());
-            if (conn.getResponseCode() < 400) {
-                size = conn.getContentLengthLong();
-            }
-            in = new ProgressMonitorInputStream(conn.getInputStream(), size, new InputStreamProgressMonitor() {
-                private int oldProgress = 0;
-
-                @Override
-                public void progress(long bytesRead, long size) {
-                    final int iProgress = (int) (bytesRead * 100 / size);
-                    if (iProgress != oldProgress) {
-                        oldProgress = iProgress;
-                        notifyProgress(uri.toASCIIString(), "Download", iProgress);
-                    }
-                }
-            });
-        } else {
-            //local file
-            notifyProgress(source, "Download", PROGRESS_MAX);
-            in = new FileInputStream(source);
-        }
-
-        return in;
+    public void addAdListener(ListenerFilmeLaden listener) {
+        listeners.add(ListenerFilmeLaden.class, listener);
     }
 
     private InputStream selectDecompressor(String source, InputStream in) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)